cot/src/cache.rs

```diff
@@ -191,8 +191,13 @@ pub type CacheResult<T> = Result<T, CacheError>;
 /// ```
 #[derive(Debug, Clone)]
 pub struct Cache {
+    inner: Arc<CacheImpl>,
+}
+
+#[derive(Debug)]
+struct CacheImpl {
     #[debug("..")]
-    store: Arc<dyn BoxCacheStore>,
+    store: Box<dyn BoxCacheStore>,
```

Comment:
Curious about this. What's the benefit of doing it this way(boxing the `store` in another struct and wrapping it in an `Arc`)? Is this cheap?